### ISSUE TITLE
deps: update cosmos-sdk to 0.45

### ISSUE THREAD

**Issue by crodriguezvega** [cosmos/ibc-go#763]
## Summary

Bump Cosmos SDK to [0.45](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.45.0).

I propose ~~to release this together with #539 and~~ to release first v1 and v2. Once we cut the final release of v3.0.0, then we cut a new minor release of v3.

____

#### For Admin Use

- [x] Not duplicate issue
- [x] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by fedekunze** [comment]
I think this is independent from #539 since the tendermint version bump wasn't included in the Cosmos SDK `v0.45` (still uses v0.34). See https://github.com/cosmos/cosmos-sdk/blob/v0.45.0/go.mod#L49

I think it would be better to release v3 with v0.45 and then update to Tendemint `v0.35` once the SDK releases a new version

**Comment by crodriguezvega** [comment]
Thanks for pointing that out, @fedekunze. We will then wait to bump Tendermint.

Regarding releasing v3 with 0.45, I was thinking that it would be better to release v3 with 0.44 since there might be chains that will perhaps take longer to update to 0.45 but still want to use interchain accounts.

**Comment by fedekunze** [comment]
We bumped to v0.45 and it didn't have additional API breaking changes. We'd need an v3 to support the new SDK release in order to launch Evmos mainnet with  a stable IBC-go release.

**Comment by crodriguezvega** [comment]
But if we release v3.1.0 with SDK 0.45 shortly after v3.0.0, wouldn't that be also ok?

**Comment by fedekunze** [comment]
yeah, that works. What's the timeline for v3.1.0? that's the only blocker we have for our mainnet

**Comment by crodriguezvega** [comment]
We arre planning to release v3.0 in the first half of March. We will do all the preparation work for bumping the SDK to 0.45, so that we can either release it together with v3.0 or release it in v3.1 right after v3.0 (and with right after I mean a matter of hours).

**Comment by colin-axner** [comment]
#701 is affected by the usage of v0.44 or v0.45. The type `MsgResponses` was included in the transaction result in SDK v0.45. Previously the type sdk.MsgData was being used as a replacement for this. Thus, if we ship interchain accounts initially with v0.44, ica auth developers would need to support decoding a slice of sdk.MsgData in addition to decoding the slice of Any MsgResponses (would depend on the version of the host chain)

This seems like an unnecessary burden since v3 is a new major version. I think it is fair to force chains to upgrade to v0.45 if they want to use interchain accounts (they'll need to modify their code anyways)

**Comment by colin-axner** [comment]
Nevermind, the SDK in-code documentation is [incorrect](https://github.com/cosmos/cosmos-sdk/blob/628547cb4e45ff880d020bebc2d02acf2ef4f221/types/abci.pb.go#L481). The usage in [v0.45](https://github.com/cosmos/cosmos-sdk/blob/b6c77e6c819f8a51166649eaef125d1bfb276f04/baseapp/baseapp.go#L755) is still the same as in v0.44

**Comment by colin-axner** [comment]
I found a way to support the format used in the upcoming v0.46 using v0.44/5 so we will continue with the plan of using v0.44 with v3

**Comment by fedekunze** [comment]
Actually, I think it would be better to support v0.45 from the beginning since it's already a major release and it has a lot of performance improvements. Users will have to upgrade no matter what and the migration from v0.44 -> v0.45 is simple and quick based on our experience.

**Comment by amaurym** [comment]
> Nevermind, the SDK in-code documentation is incorrect. The usage in v0.45 is still the same as in v0.44

That's correct, sorry about that, it's a protobuf comment that we forgot to update when releasing v0.45. It's tracked on the SDK now.

**Comment by colin-axner** [comment]
We decided we will release v0.45 in v3

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
